app/models/asset.rb

```diff
@@ -85,6 +85,11 @@ class Asset
   scope :undeleted, -> { where(deleted_at: nil) }
 
   state_machine :state, initial: :unscanned do
+    around_transition do |asset, transition, block|
+      Rails.logger.info("#{asset.id} - Asset#state_machine - event: #{transition.event}")
```

Comment:
I'm not familiar with this state machine, so I'll assume that all Assets have an `#id` for all transitions (i.e. there's no point in using safe navigation here)

Reply:
(And I see that `Transition#event` comes from https://github.com/state-machines/state_machines/blob/v0.6.0/lib/state_machines/transition.rb#L65, so that's nothing to worry about)